crates/matrix-sdk-ui/src/timeline/event_handler.rs

```diff
@@ -1297,6 +1272,128 @@ impl<'a, 'o> TimelineEventHandler<'a, 'o> {
         }
     }
 
+    /// Remove the local timeline item matching the `event_id` or the
+    /// `transaction_id` of `new_event_timeline_item` if it exists.
+    ///
+    /// Let's also try to deduplicate remote evevents. If `VectorDiff`s are the
+    /// inputs of the `Timeline`, this is not necessary, as they are
+    /// generated by the `EventCache`, which supports its own deduplication
+    /// algorithm.
+    //
+    // Note: this method doesn't take `&mut self` to avoid a borrow checker
+    // conflict with `TimelineEventHandler::add_item`. It's fine.
```

Comment:
The "It's fine" feels very very coercitive, here and below :sweat_smile: :sweat_smile: :sweat_smile:

Reply:
I knew you would grimace ^^.